### PR TITLE
Add nba_cup task

### PR DESCRIPTION
**Pull request by kmandziy**

### CODE REVIEW

**Review by AlinaKovtun — APPROVED** [review]

File: Tasks/nba_cup.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+#returns "Team Name:This team didn't play!".
+
+#If a score is a float number will 
+#returns "Error(float number):the concerned string"
+
+#If the team name "" returns ""
+
+def nba_cup(result_sheet, to_find)
+  return "" if to_find.empty?
+  w = a = d = l = scored = conceded = 0
```
Comment (omaik) [inline Tasks/nba_cup.rb]: one letter names are very poor. It is hard to understand what they mean.

File: Tasks/nba_cup.rb [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  team_exist = false
+  
+  result_sheet.split(',').each do |game|
+    teams = game.split(/(?!\d{1,3}[a-z]{1,3})(?:\d+\.+\d{1,3}|\d{1,3})/).map(&:strip)
+    score = game.scan(/(?!\d{1,3}[a-z]{1,3})(?:\d+\.+\d{1,3}|\d{1,3})/)
```
Comment (omaik) [inline Tasks/nba_cup.rb]: Rewrite this to parse all data in one regular expression. It should be much simpler in that way.

File: Tasks/nba_cup.rb (hunk 1 above)
Comment (omaik) [reply Tasks/nba_cup.rb]: I may be blind, but seems that varialbe `a` is not used at all.

File: Tasks/nba_cup.rb [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    w += 1 if to_find_score > enemy_score
+    l += 1 if to_find_score < enemy_score
+    d += 1 if to_find_score == enemy_score 
+    conceded += enemy_score
+    scored += to_find_score
+    team_exist = true
+  end  
+  
+  return "#{to_find}:This team didn't play!" unless team_exist
+  points = w*3 + d
```
Comment (omaik) [inline Tasks/nba_cup.rb]: whitespaces

File: Tasks/nba_cup.rb [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+#If a team name can't be found in the given string 
+#returns "Team Name:This team didn't play!".
+
+#If a score is a float number will 
+#returns "Error(float number):the concerned string"
+
+#If the team name "" returns ""
+
+def nba_cup(result_sheet, to_find)
```
Comment (omaik) [inline Tasks/nba_cup.rb:75]: this method could be split into logical methods. Each responsible for its own part. 1. Parsing Input 2. Processing data. 3. Presenting output.

**Review by PidtserkovnyiVitalii — APPROVED** [review]

**Pull request merged**